mathesar_ui/src/sections/table-view/actions-pane/record-operations/FilterEntry.svelte

```diff
@@ -0,0 +1,243 @@
+    clearTimeout(timer);
+    timer = window.setTimeout(() => {
+      onValueUpdate(_value);
+    }, 500);
+  }
+  $: onValueChange(value);
+
+  function onValueChangeFromUser() {
+    clearTimeout(timer);
+    onValueUpdate(value);
+  }
+
+  function calculateInputCap(
+    _selectedCondition?: AbstractTypeFilterDefinition,
+    _processedSelectedColumn?: ProcessedTableColumn,
+  ):
+    | { column: CellColumnLike }
+    | { componentAndProps: ComponentAndProps<unknown> }
+    | undefined {
+    const parameterTypeId = _selectedCondition?.parameters[0];
+    // If there are no parameters, show no input. eg., isEmpty
+    if (typeof parameterTypeId === 'undefined') {
+      return undefined;
+    }
+
+    // If there is a parameter
+    // Check if the type is same as column's type.
+    // If yes, pass down column's calculated cap.
+    // If no, pass down the type directly.
+    const abstractTypeId =
+      _processedSelectedColumn?.abstractTypeOfColumn.identifier;
+    if (abstractTypeId === parameterTypeId && selectedColumnInputCap) {
+      return { componentAndProps: selectedColumnInputCap };
+    }
+    return {
+      column: {
+        type: parameterTypeId,
+        type_options: {},
+        display_options: {},
+      },
+    };
+  }
+
+  $: inputCap = calculateInputCap(selectedCondition, processedSelectedColumn);
+
+  function onColumnChange() {
+    prevValue = undefined;
+    value = undefined;
+    dispatch('update');
+  }
+
+  function onConditionChange(_conditionId?: string) {
+    if (!_conditionId) {
+      return;
+    }
+    const condition = selectedColumnFiltersMap.get(_conditionId);
+    if (!condition) {
+      return;
+    }
+    if (typeof condition?.parameters[0] === 'undefined') {
+      prevValue = undefined;
+      value = undefined;
+    }
+    dispatch('update');
+  }
+</script>
+
+<div class="filter-entry">
+  <div class="prefix">
+    <slot />
+  </div>
+  <InputGroup>
+    <Select
+      options={columnIds}
+      bind:value={columnId}
+      getLabel={getColumnName}
+      on:change={onColumnChange}
+      triggerClass="filter-column-id"
+    />
+    <Select
+      options={conditionIds}
+      bind:value={conditionId}
+      getLabel={getConditionName}
+      on:change={(e) => onConditionChange(e.detail)}
+      triggerClass="filter-condition"
+    />
+    {#if inputCap}
+      <DataTypeBasedInput
+        bind:value
+        {...inputCap}
+        on:input={() => {
+          showError = true;
+        }}
+        on:blur={() => {
+          showError = true;
+        }}
+        on:change={onValueChangeFromUser}
+        class="filter-input"
+        hasError={showError && !isValid}
+      />
+    {/if}
+    <Button
+      size="small"
+      class="filter-remove"
+      on:click={() => dispatch('removeFilter')}
+    >
+      <Icon data={faTrashAlt} />
+    </Button>
+  </InputGroup>
+</div>
+
+<style lang="scss">
+  .filter-entry {
+    display: flex;
+    min-width: 560px;
```

Comment:
It seems a lot of this CSS is new. Is there a reason you've used `px` instead of `em` or `rem`, per our [standard](https://wiki.mathesar.org/en/engineering/standards/frontend#css-units)?

Reply:
I'm treating most of this css as temporary since we probably might have to re-work some/most of the filtering, sorting and grouping when working on views. I used px to finish this quick. I'm waiting for the UI milestone to work purely on the styles in the entire product.